test/integration/sds_dynamic_integration_test.cc

```diff
@@ -364,18 +458,21 @@ class SdsDynamicDownstreamCertValidationContextTest : public SdsDynamicDownstrea
                                    ->mutable_listeners(0)
                                    ->mutable_filter_chains(0)
                                    ->mutable_transport_socket();
-      envoy::extensions::transport_sockets::tls::v3::DownstreamTlsContext tls_context;
-      auto* common_tls_context = tls_context.mutable_common_tls_context();
-      common_tls_context->add_alpn_protocols(Http::Utility::AlpnNames::get().Http11);
-
-      auto* tls_certificate = common_tls_context->add_tls_certificates();
-      tls_certificate->mutable_certificate_chain()->set_filename(
-          TestEnvironment::runfilesPath("test/config/integration/certs/servercert.pem"));
-      tls_certificate->mutable_private_key()->set_filename(
-          TestEnvironment::runfilesPath("test/config/integration/certs/serverkey.pem"));
-      setUpSdsValidationContext(common_tls_context);
-      transport_socket->set_name("envoy.transport_sockets.tls");
-      transport_socket->mutable_typed_config()->PackFrom(tls_context);
+      if (!test_quic_) {
+        envoy::extensions::transport_sockets::tls::v3::DownstreamTlsContext tls_context;
+        auto* common_tls_context = tls_context.mutable_common_tls_context();
+        configInlinedCerts(common_tls_context);
+        transport_socket->set_name("envoy.transport_sockets.tls");
+        transport_socket->mutable_typed_config()->PackFrom(tls_context);
+      } else {
+        envoy::extensions::transport_sockets::quic::v3::QuicDownstreamTransport
+            transport_socket_config;
+        auto* common_tls_context =
+            transport_socket_config.mutable_downstream_tls_context()->mutable_common_tls_context();
+        configInlinedCerts(common_tls_context);
+        transport_socket->set_name("envoy.transport_sockets.quic");
+        transport_socket->mutable_typed_config()->PackFrom(transport_socket_config);
```

Comment:
I feel like we're getting the quic vs tls stuff in a lot of places and I'd like to reduce that.  Can you think of something cleaner?  Passing in the context to a lambda and having it fill in the right one?  Setting the h2 tls context by default and having a convertIfHttp3 function which moved it over and set the correct transport socket name?

Reply:
I added a utility function in ConfigHelper to distinguish ssl v.s quic.

Reply:
Should we use the new test helper here?